### PR TITLE
test: SpanEvent & ServiceCall ViewGenerator

### PR DESCRIPTION
**Pull request by rish691**
Minor code refactoring and tests
No logical changes

### CODE REVIEW

File: hypertrace-view-generator/hypertrace-view-generator/src/main/java/org/hypertrace/viewgenerator/generators/ServiceCallViewGenerator.java [hunk 1]
```diff
@@ -62,156 +62,127 @@
       Map<String, Entity> entityMap,
       Map<ByteBuffer, Event> eventMap,
       Map<ByteBuffer, List<ByteBuffer>> parentToChildrenEventIds,
-      Map<ByteBuffer, ByteBuffer> childToParentEventIds) {
+      Map<ByteBuffer, ByteBuffer> childToParentEventId) {
```
Comment (tim-mwangi) [inline hypertrace-view-generator/hypertrace-view-generator/src/main/java/org/hypertrace/viewgenerator/generators/ServiceCallViewGenerator.java:65]: nit: `childToParentEventIdMap`

**Review by tim-mwangi — APPROVED** [review]

File: hypertrace-view-generator/hypertrace-view-generator/src/test/resources/StructuredTrace-Hotrod.avro [hunk 2]
```diff
@@ -0,0 +1,158 @@
...e common between the two nodes. request/response  e.g. api_name, url etc","default":null},{"name":"start_time_millis","type":["null","long"],"default":null},{"name":"end_time_millis","type":["null","long"],"default":null},{"name":"metrics","type":["null","Metrics"],"default":null}]}}},{"name":"event_edge_list","type":{"type":"array","items":"Edge"}},{"name":"entity_event_edge_list","type":{"type":"array","items":"Edge"}},{"name":"entity_entity_graph","type":["null",{"type":"record","name":"Graph","fields":[{"name":"root_nodes","type":{"type":"array","items":"int"},"default":[]},{"name":"adjacency_list","type":{"type":"array","items":{"type":"array","items":"int"}},"default":[]}]}],"default":null},{"name":"event_event_graph","type":["null","Graph"],"default":null},{"name":"entity_event_graph","type":["null","Graph"],"default":null},{"name":"timestamps","type":["null",{"type":"record","name":"Timestamps","fields":[{"name":"records","type":{"type":"map","values":{"type":"record","name":"TimestampRecord","fields":[{"name":"name","type":{"type":"string","avro.java.string":"String"}},{"name":"timestamp","type":"long"}]},"avro.java.string":"String"},"default":{}}]}],"default":null}]} �H��
```
Comment (kotharironak) [inline hypertrace-view-generator/hypertrace-view-generator/src/test/resources/StructuredTrace-Hotrod.avro:1]: Can you add JSON for this too as in one place you have added both avro and json file?

File: hypertrace-view-generator/hypertrace-view-generator/src/main/java/org/hypertrace/viewgenerator/generators/ServiceCallViewGenerator.java [hunk 3]
```diff
@@ -62,156 +62,127 @@
...
-                rootEvent ->
-                    rootEvent
-                        .getEventRefList()
-                        .isEmpty()) // Real roots and not broken spans with a parent span Id whose
-            // corresponding span object is not part of the trace.
-            .map(rootEvent -> createViewForNonEntryRootSpan(structuredTrace, rootEvent))
-            .collect(Collectors.toUnmodifiableList());
-
-    serviceCallViewRecords.addAll(serviceCallViewsForExitsThatAreRoots);
+    // Scenario #4: Handle non entry roots
```
Comment (kotharironak) [inline hypertrace-view-generator/hypertrace-view-generator/src/main/java/org/hypertrace/viewgenerator/generators/ServiceCallViewGenerator.java:78]:
```suggestion
    // Scenario #4: Handle non entry roots (broker trace)
```

File: hypertrace-view-generator/hypertrace-view-generator/src/test/java/org/hypertrace/viewgenerator/generators/SpanEventViewGeneratorTest.java [hunk 4]
```diff
@@ -89,4 +101,36 @@ public void testGetRequestUrl_urlAndPathIsAbsent() {
...
+  private void verifyCreateExitSpanToApiEntrySpan(
+      StructuredTrace trace, TraceState traceState) {
+    Map<ByteBuffer, Event> exitSpanToApiEntrySpanMap =
+        spanEventViewGenerator.getExitSpanToCalleeApiEntrySpanMap(
+            trace.getEventList(), traceState.getChildToParentEventIds(),
+            traceState.getParentToChildrenEventIds(), traceState.getEventMap());
+
+    // verify for all entries in the map, key is exit span and value is entry api boundary
+    exitSpanToApiEntrySpanMap.forEach((key, value) -> {
+      EnrichedSpanUtils.isExitSpan(traceState.getEventMap().get(key));
```
Comment (kotharironak) [inline hypertrace-view-generator/hypertrace-view-generator/src/test/java/org/hypertrace/viewgenerator/generators/SpanEventViewGeneratorTest.java:132]: Wrapped around AssertTrue

File: hypertrace-view-generator/hypertrace-view-generator/src/test/java/org/hypertrace/viewgenerator/generators/SpanEventViewGeneratorTest.java [hunk 5]
```diff
@@ -89,4 +101,36 @@ public void testGetRequestUrl_urlAndPathIsAbsent() {
...
+      EnrichedSpanUtils.isEntryApiBoundary(value);
```
Comment (kotharironak) [inline hypertrace-view-generator/hypertrace-view-generator/src/test/java/org/hypertrace/viewgenerator/generators/SpanEventViewGeneratorTest.java:133]: Wrapped around AssertTrue

**Review by kotharironak — APPROVED** [review]
LGTM, minor comments

**Comment by rish691** [comment]
Will take care of comments in subsequent change.

**Pull request merged**